src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -678,8 +687,31 @@ public void Commit(bool previewChanges = false)
         /// <returns><see langword="true"/> if the rename operation was commited, <see
         /// langword="false"/> otherwise</returns>
         private bool CommitWorker(bool previewChanges)
+        {
+            return _threadingContext.JoinableTaskFactory.Run(() => CommitWorkerAsync(previewChanges));
+        }
+
+        public Task CommitAsync(bool previewChanges = false)
+            => CommitWorkerAsync(previewChanges);
+
+        private async Task<bool> CommitWorkerAsync(bool previewChanges)
         {
             _threadingContext.ThrowIfNotOnUIThread();
+
+            var indicatorFactory = _workspace.Services.GetRequiredService<IBackgroundWorkIndicatorFactory>();
+
+            // We do not cancel on edit because as part of the rename system we have asynchronous work still occurring
+            // that itself may be asynchronously editing the buffer (for example, updating reference locations with the
+            // final renamed text).  Ideally though, once we start comitting, we would cancel any of that work and then
+            // only have the work of rolling back to the original state of the world and applying the desired edits
+            // ourselves.
+            using var context = indicatorFactory.Create(
+                _triggerView, _triggerSpan, EditorFeaturesResources.Computing_Rename_information,
+                cancelOnEdit: false, cancelOnFocusLost: false);
```

Comment:
hmm... so could you type something after triggering it, the rename continues, then but have the rename then fail to apply (or partially apply) because you typed and now tryapply fails?  How do I know when that happens?

Seems risky

Reply:
good point.  need to block this PR on figuring that out.